base/ca/src/main/java/org/dogtagpki/server/ca/rest/v2/CertServlet.java

```diff
@@ -107,7 +107,7 @@ public void getCert(HttpServletRequest request, HttpServletResponse response) th
         }
     }
 
-    @WebAction(method = HttpMethod.POST, paths = {"search"})
+    @WebAction(method = HttpMethod.POST, paths = {""})
```

Comment:
The general REST semantic for POST is to create something or modify something in the server and it is generally used across the APIs so if we post to `/ca/certs` I am expecting to generate a cert entry so I would not modify this path

Reply:
I've reverted the path changes for now, we can revisit this issue another time.

My concern with this path is that the `search` keyword in `POST /ca/v2/certs/search` occupies the same position as the cert ID in `GET /ca/v2/certs/{id}`. Although they use different HTTP methods, it doesn't look RESTful since `/ca/v2/certs/search` does not point to an actual object.

For creating/adding an object I think usually it's advised to use `PUT`. `POST` is kind of a generic method that can be used for anything that doesn't fit other HTTP methods. Maybe it would be better to use something like `POST /ca/v2/certs?action=search`.

Reply:
> For creating/adding an object I think usually it's advised to use `PUT`. `POST` is kind of a generic method that can be used for anything that doesn't fit other HTTP methods. Maybe it would be better to use something like `POST /ca/v2/certs?action=search`.

Actually, POST is the method to create a new resource. PUT is used to create or modify a resource where you already know the ID (these some example from a very quick search: [restfulapi web](https://restfulapi.net/http-methods/), [mozilla](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) or [geeksforgeeks](https://www.geeksforgeeks.org/5-http-methods-in-restful-api-development/)).

Reply:
In our case the certs are always generated by the CA, so the client will never call this API to create a cert (enrollment/renewal is done using a different interface and it takes a CSR instead of a cert). Suppose we change the API to allow an authorized user to upload an existing cert to the CA, we might use `PUT /ca/v2/certs/<cert ID>`. If we don't want to require the client to include the cert ID in the request we probably can use `POST`, but I'd suggest we use `POST /ca/v2/certs?action=upload`. That way we can still use `POST /ca/v2/certs?action=search` for search.

Generally I'd say that all REST resources should support a `GET` operation. If `GET` doesn't make sense for that resource, like `GET /ca/v2/certs/search`, it's not really a RESTful design.